pkgs/development/compilers/rust/make-rust-platform.nix

```diff
@@ -17,59 +18,64 @@
   ...
 }:
 
-rec {
-  fetchCargoTarball = buildPackages.callPackage ../../../build-support/rust/fetch-cargo-tarball {
-    git = buildPackages.gitMinimal;
-    inherit cargo;
-  };
+(makeScopeWithSplicing' {
+  otherSplices = generateSplicesForMkScope "rustPlatform";
```

Comment:
This locks down `makeRustPlatform` so it can only be used as `pkgs.rustPlatform`.

If you define your own custom platform with `makeRustPlatform`, then `rustc` in other splices will point to `pkgs.rustPlatform` which is incorrect.

Reply:
I think the correct thing to do here would be either:
a) Pass down `otherSplices` to the caller of `makeRustPlatform`
b) Build all `otherSplices` manually (`selfBuildBuild = lib.makeScope pkgsBuildBuild.newScope f`, ...)